forge-game/src/main/java/forge/game/ability/effects/CopyPermanentEffect.java

```diff
@@ -297,6 +306,13 @@ public static Card getProtoType(final SpellAbility sa, final Card original, fina
         } else {
             copy.setState(copy.getCurrentStateName(), true, true);
         }
+        if (sa.hasParam("DefinedName")) {
```

Comment:
in this case you can just do `copy = original` above instead of another CardFactory?

Reply:
like this?